graph/input.go

```diff
@@ -130,32 +171,70 @@ func (ip *InputProcessor[T]) process2D(v [][]T, kernel *kernel.Kernel) error {
 		return err
 	}
 
-	switch ip.dtype {
-	case tensor.Float:
+	switch v := any(v).(type) {
```

Comment:
I doubt that 2D map types exist. There isn't any need for processing the 2D variants.

Reply:
yeah. This makes sense. will update